server/types/app.go

```diff
@@ -37,6 +38,10 @@ type (
 		// RegisterGRPCServer registers gRPC services directly with the gRPC
 		// server.
 		RegisterGRPCServer(grpc.Server)
+
+		// RegisterTxService registers the gRPC Query service for tx (such as tx
+		// simulation, fetching txs by hash...).
+		RegisterTxService(clientCtx client.Context)
```

Comment:
To make a call to tendermint, I need a TM `rpcclient.Client`, right? Or maybe there's a way to do it directly via baseapp?

For now, I went with the TM `rpcclient.Client`, but it requires an client (e.g. the Validator in our network config) to manually `RegisterTxService` with its client context.